numpy/core/overrides.py

```diff
@@ -148,6 +159,13 @@ def array_function_dispatch(dispatcher, module=None, verify=True,
         If True, copy docs from the dispatcher function onto the dispatched
         function, rather than from the implementation. This is useful for
         functions defined in C, which otherwise don't have docstrings.
+    function: callable or None
+        If function is callable, it will replace the function passed as
+        implementation. This is useful when the function being decorated
+        is just a helper and not the de-facto implementation with the
+        expected signature, particularly its name. Usually used for the
+        like= parameter to avoid decorating the actual implementation due
```

Comment:
@pentschev thanks! Two quick questions/thoughts (I don't think I will give it a thorough pass today):

1.  Should we rename `function=` into `public_api=`?
2. Passing this as `implementation` is fine, but as currently implemented I think only for functions taking `like=` which gets removed so that recursive calls are not problematic.  I am OK with keeping it as is, but maybe change the doc from "Usually used..." into "Currently only supported for the ``like=`` parameter"

Reply:
Agreed. At first I didn't use `public_api` because it's already used by the decorator internally, which causes issues with having the same name. I now renamed the internal one to `_public_api`, if anyone objects to that, please let me know and I'll change it back and we'll try to find a better name for the kwarg that doesn't conflict with any internals.

I've addressed both points in https://github.com/numpy/numpy/pull/16935/commits/e00c33be24303868462e4ff29fc4c5e3f80115bf .